### DISCUSSION TITLE
userdma接線問題

### DISCUSSION THREAD

**Discussion by DonghanTsai** [bol-edu/HLS-SOC-Discussions#221] [General]
在lab4-1中，userdma的m_axi_gmem0與m_axi_gmem1原本分別會接到axi_smc_1與axi_smc_2，但是因為4-2沒有這些vip了，所以也就沒有axi_smc_1與axi_smc_2這兩個block，故想請問在4-3是要自己再把這兩個也重新創一個出來嗎，如果是的話，那在創出來之後我們要如何把data先寫進去?(4-1是用fsic_tb.v寫但4-2也沒有tb了)，如果不是的話那我們應該如何去接userdma的這兩個protocal呢?

**Comment by nthuyouwei** [comment]
開啟Vivado,把userdma 加入後按run auto就會幫你處理完gmem0 gmem1，只有兩條instream跟outstream需要自己手動接到userdmaso跟userdmasi ，啊那些對應的config 地址可能要自己開Vivado看，不會像simulation 一樣在0x9000

**Comment by nthuyouwei** [comment]
這時候不需要tb,而是要上版子跑Python,可以參考4-2去做修改！

**Reply by DonghanTsai** [reply]
我按照指示連線並產生bitstream了，不過在跑ipynb的時候發現寫userdma config時kernel會直接死掉，請問我這樣子寫的話有設定錯嗎? (圖一是vivado的address map，可以看到userdma的位子應該是在4002_0000，圖2是我在ipynb中自己加的程式碼)
![螢幕擷取畫面 2024-05-11 201126](https://github.com/bol-edu/HLS-SOC-Discussions/assets/71302677/10117f4e-2b72-4388-a00e-fce35d6003d6)
![螢幕擷取畫面 2024-05-11 203436](https://github.com/bol-edu/HLS-SOC-Discussions/assets/71302677/b00f72dd-3728-4928-95c8-72a1951d9b49)

**Comment by nthuyouwei** [comment]
這部分看起來沒錯，因為我也是類似的寫法。

**Comment by michael61112** [comment]
在vivado內需要卻設定userDMA的address segment才能去control
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/9332860/b9a30a65-2bfb-4dec-8faa-ea4018439bf5)

PS side 我是用0x9000 去control userDMA
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/9332860/fdfb96c1-c3ed-4850-ab28-7e8924620f56)
